### PR TITLE
Upload HLS files to S3 on the fly

### PR DESCRIPTION
**Pull request by mustafaboleken**
https://github.com/ant-media/Ant-Media-Server/issues/5082

### CODE REVIEW

File: src/main/java/io/antmedia/servlet/UploadHLSChunk.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+			ConfigurableWebApplicationContext appContext = (ConfigurableWebApplicationContext) req.getServletContext().getAttribute(WebApplicationContext.ROOT_WEB_APPLICATION_CONTEXT_ATTRIBUTE);
+
+			if (appContext != null && appContext.isRunning())
+			{
+				String applicationName = appContext.getApplicationName();
+
+				String filepath = WEBAPPS + applicationName + STREAMS + req.getPathInfo();
+
+				InputStream fileContent = req.getInputStream();
+				File targetFile = new File(filepath + ".tmp");
```
[the remark this answers is not in the document]
Comment (mustafaboleken) [reply src/main/java/io/antmedia/servlet/UploadHLSChunk.java:127]: @sonatype-lift ignore

File: src/main/java/io/antmedia/servlet/UploadHLSChunk.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+		} catch (Exception e) {
+			logger.error(ExceptionUtils.getStackTrace(e));
+		}
+
+	}
+
+	public boolean checkIfStreamIsFinished(String filePath) {
+		boolean isFinished = false;
+		try {
+			File file = new File(filePath);
```
[the remark this answers is not in the document]
Comment (mustafaboleken) [reply src/main/java/io/antmedia/servlet/UploadHLSChunk.java:175]: @sonatype-lift ignore

File: src/main/java/io/antmedia/AntMediaApplicationAdapter.java [hunk 3]
```diff
@@ -1612,6 +1612,7 @@ public void setStorageclientSettings(AppSettings settings) {
 		storageClient.setEnabled(settings.isS3RecordingEnabled());
 		storageClient.setPermission(settings.getS3Permission());
 		storageClient.setStorageClass(settings.getS3StorageClass());
+		storageClient.setCacheControl(settings.getS3CacheControl());
```
[the remark this answers is not in the document]
Comment (mustafaboleken) [reply src/main/java/io/antmedia/AntMediaApplicationAdapter.java:1615]: @sonatype-lift ignore

File: src/main/java/io/antmedia/servlet/UploadHLSChunk.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mekya) [reply src/main/java/io/antmedia/servlet/UploadHLSChunk.java]: @sonatype-lift ignoreall

**Pull request merged**